include/internal/common.h

```diff
@@ -93,87 +93,170 @@ __owur static inline int ossl_assert_int(int expr, const char *exprstr,
+     l |= ((unsigned long)(*((c)++)))       )
+
+# define l2n(l, c) \
+    (*((c)++) = (unsigned char)(((l) >> 24) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 16) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >>  8) & 0xff), \
+     *((c)++) = (unsigned char)(((l)      ) & 0xff))
+
+# define n2l8(c, l) \
+    (l  = ((uint64_t)(*((c)++))) << 56, \
+     l |= ((uint64_t)(*((c)++))) << 48, \
+     l |= ((uint64_t)(*((c)++))) << 40, \
+     l |= ((uint64_t)(*((c)++))) << 32, \
+     l |= ((uint64_t)(*((c)++))) << 24, \
+     l |= ((uint64_t)(*((c)++))) << 16, \
+     l |= ((uint64_t)(*((c)++))) <<  8, \
+     l |= ((uint64_t)(*((c)++)))      )
+
+# define l2n8(l, c) \
+    (*((c)++) = (unsigned char)(((l) >> 56) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 48) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 40) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 32) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 24) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 16) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >>  8) & 0xff), \
+     *((c)++) = (unsigned char)(((l)      ) & 0xff) )
+
+# define c2ln(c, l1, l2, n) \
+    { \
+        c += n; \
+        l1 = l2 = 0; \
+        switch (n) { \
+        case 8: l2  = ((unsigned long)(*(--(c)))) << 24; \
+                /* fallthrough */ \
+        case 7: l2 |= ((unsigned long)(*(--(c)))) << 16; \
+                /* fallthrough */ \
+        case 6: l2 |= ((unsigned long)(*(--(c)))) <<  8; \
+                /* fallthrough */ \
+        case 5: l2 |= ((unsigned long)(*(--(c))))      ; \
+                /* fallthrough */ \
+        case 4: l1  = ((unsigned long)(*(--(c)))) << 24; \
+                /* fallthrough */ \
+        case 3: l1 |= ((unsigned long)(*(--(c)))) << 16; \
+                /* fallthrough */ \
+        case 2: l1 |= ((unsigned long)(*(--(c)))) <<  8; \
+                /* fallthrough */ \
+        case 1: l1 |= ((unsigned long)(*(--(c))))      ; \
+        } \
+    }
+
+# define l2cn(l1, l2, c, n) \
+    { \
+        c += n; \
+        switch (n) { \
+        case 8: *(--(c)) = (unsigned char)(((l2) >> 24) & 0xff); \
+                /* fallthrough */ \
+        case 7: *(--(c)) = (unsigned char)(((l2) >> 16) & 0xff); \
+                /* fallthrough */ \
+        case 6: *(--(c)) = (unsigned char)(((l2) >>  8) & 0xff); \
+                /* fallthrough */ \
+        case 5: *(--(c)) = (unsigned char)(((l2)      ) & 0xff); \
+                /* fallthrough */ \
+        case 4: *(--(c)) = (unsigned char)(((l1) >> 24) & 0xff); \
+                /* fallthrough */ \
+        case 3: *(--(c)) = (unsigned char)(((l1) >> 16) & 0xff); \
+                /* fallthrough */ \
+        case 2: *(--(c)) = (unsigned char)(((l1) >>  8) & 0xff); \
+                /* fallthrough */ \
+        case 1: *(--(c)) = (unsigned char)(((l1)      ) & 0xff); \
+        } \
+    }
+
+#define n2ln(c, l1, l2, n) \
+    { \
+        c += n; \
+        l1 = l2 = 0; \
```

Comment:
Missing parentheses in the previous two lines (and all the `l1` and `l2` below).

Reply:
fixed